Monika After Story/game/script-holidays.rpy

```diff
@@ -4402,11 +4405,44 @@ label mas_player_bday_autoload_check:
             and not mas_isO31()
             and not mas_isF14()
         ):
-        # starting player b_day off with a closed door greet
-        $ mas_skip_visuals = True
-        $ selected_greeting = "i_greeting_monikaroom"
-        # need this so we don't get any strange force quit dlg after the greet
-        $ persistent.closed_self = True
+
+        python:
+            # first we determine if we want to run a surprise greeting this year
+            this_year = datetime.date.today().year
+            years_checked = range(this_year-10,this_year)
+            times_ruined = 0
+            surp_int = 3
+
+            for y in years_checked:
+                if y in mas_HistVerify("player_bday.opened_door",True)[1]:
```

Comment:
so `mas_HistVerify` can be run with years. either:
* `mas_HistVerify("player_bday.opened_door", True, *years_checked)`

OR
* `mas_HistVerify_k(years_checked, True, "player_bday", "opened_door")`

the 2nd element of the returned tuple is list of years that matched verification so you can just do a `len` on the returned list as times_ruined instead of doing an extra loop here.